src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx

```diff
@@ -32,31 +38,156 @@ const pageKey = 'vaFacilityV2';
 const pageTitle = 'Choose a VA location for your appointment';
 
 function VAFacilityPageV2({
-  schema,
+  canScheduleAtChosenFacility,
+  childFacilitiesStatus,
   data,
+  eligibility,
   facilities,
-  childFacilitiesStatus,
+  facility,
+  facilityDetails,
+  facilityDetailsStatus,
+  loadingEligibilityStatus,
+  noValidVAParentFacilities,
+  noValidVAFacilities,
   openFacilityPageV2,
-  updateFacilityPageV2Data,
-  typeOfCare,
-  routeToPreviousAppointmentPage,
   pageChangeInProgress,
+  parentDetails,
+  parentFacilitiesStatus,
+  routeToPreviousAppointmentPage,
+  routeToNextAppointmentPage,
+  schema,
+  singleValidVALocation,
+  siteId,
+  typeOfCare,
+  updateFacilityPageV2Data,
 }) {
-  useEffect(() => {
-    document.title = `${pageTitle} | Veterans Affairs`;
-    scrollAndFocus();
-    openFacilityPageV2(pageKey, uiSchema, initialSchema);
-  }, []);
-
-  if (childFacilitiesStatus === FETCH_STATUS.loading) {
-    return <LoadingIndicator message="Finding locations" />;
+  const history = useHistory();
+  const loadingEligibility = loadingEligibilityStatus === FETCH_STATUS.loading;
+  const loadingParents = parentFacilitiesStatus === FETCH_STATUS.loading;
+  const loadingFacilities = childFacilitiesStatus === FETCH_STATUS.loading;
+  const [showEligibilityModal, setShowEligibilityModal] = useState(false);
+
+  useEffect(
+    () => {
+      document.title = `${pageTitle} | Veterans Affairs`;
+      scrollAndFocus();
+      openFacilityPageV2(pageKey, uiSchema, initialSchema);
+    },
+    [openFacilityPageV2],
+  );
+
+  useEffect(
+    () => {
+      if (!loadingEligibility && canScheduleAtChosenFacility === false) {
+        setShowEligibilityModal(true);
+      }
+    },
+    [loadingEligibility, canScheduleAtChosenFacility],
+  );
```

Comment:
Sets `showEligibilityModal` to `true` if eligibility has loaded and user can not schedule at chosen facility